src/scenes/Authentication/Login/Login.tsx

```diff
@@ -27,9 +27,10 @@ export const Login = (props: Except<Props, 'onSubmit'>) => {
         variables: { input },
         update: (cache, { data }) => {
           const user = data?.login.user;
-          if (user) {
+          const powers = data?.login.powers;
+          if (user || powers) {
             setSuccess(true);
-            updateSessionCache(cache, user);
+            updateSessionCache(cache, user, powers);
           }
```

Comment:
Just a small thing but maybe `updateSessionCache` could take a session-like object instead `{ user, powers }`.

```tsx
          const session = data?.login;
          if (session) {
            setSuccess(true);
            updateSessionCache(cache, session);
          }
```

Reply:
It can...althought technically `login` doesn't have to match session, so inside the function we'd have to do the destructuring

Reply:
How does it not match?
```tsx
const updateSessionCache = (cache, newSession: { user: User, powers: Power[] }) => {};
```

Reply:
They match now because we made the fragments match (we had to manually add the powers field on the login and register queries), but they are still separate fragments.  But currently yes they are the same data.

I'm not even sure we're concerned about the same thing, but yes I can pass in this session like object, it's cleaner

Reply:
Sure I didn't mean change the gql fragments.